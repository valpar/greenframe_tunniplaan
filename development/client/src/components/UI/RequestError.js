import content from "../../assets/content/content.json";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faExclamation } from "@fortawesome/free-solid-svg-icons";

export const RequestError = (props) => {
  const { errorMessage, requestHandler, size, buttonText } = props;

  return (
    <div
      className={`p-4 lg:mt-3 border border-borderGray shadow shadow-borderGray ${
        size ? size : "w-full"
      }`}
    >
      <div className="flex justify-center py-4 text-4xl lg:text-6xl text-collegeRed">
        <FontAwesomeIcon icon={faExclamation} />
      </div>
      <p className="pb-4">
        {errorMessage ? errorMessage : content.errorMessages.serverError}
      </p>
      <button
        onClick={requestHandler}
        className="py-1 px-8 my-4 border border-borderGray shadow hover:bg-borderGray hover:shadow-lg duration-150"
      >
        {buttonText ? buttonText : "Uuesti"}
      </button>
    </div>
  );
};
