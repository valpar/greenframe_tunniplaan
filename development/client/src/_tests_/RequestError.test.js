import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { RequestError } from '../components/UI/RequestError';

describe('RequestError', () => {
  it("matches snapshot", () => {
    const tree = renderer.create(<RequestError />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('displays the custom error message', () => {
    const customMessage = "Custom Error Message";
    render(<RequestError errorMessage={customMessage} />);
    expect(screen.getByText(customMessage)).toBeInTheDocument();
  });
  it('displays default error message when no custom message is provided', () => {
    render(<RequestError />);
    // Replace 'Default Error Message' with the actual default message from your content file
    expect(screen.getByText('Andmete laadimine ebaÃµnnestus!')).toBeInTheDocument();
  });
  it('calls reloadHandler on button click', () => {
    const mockReloadHandler = jest.fn();
    render(<RequestError reloadHandler={mockReloadHandler} />);
    fireEvent.click(screen.getByText('Uuesti'));
    expect(mockReloadHandler).toHaveBeenCalled();
  });
});