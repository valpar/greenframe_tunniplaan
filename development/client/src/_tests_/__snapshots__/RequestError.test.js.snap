// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestError matches snapshot 1`] = `
<div
  className="p-4 lg:mt-3 border border-borderGray shadow shadow-borderGray w-full"
>
  <div
    className="flex justify-center py-4 text-4xl lg:text-6xl text-collegeRed"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-exclamation "
      data-icon="exclamation"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 128 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M96 64c0-17.7-14.3-32-32-32S32 46.3 32 64V320c0 17.7 14.3 32 32 32s32-14.3 32-32V64zM64 480c22.1 0 40-17.9 40-40s-17.9-40-40-40s-40 17.9-40 40s17.9 40 40 40z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
  <p
    className="pb-4"
  >
    Andmete laadimine eba√µnnestus!
  </p>
  <button
    className="py-1 px-8 my-4 border border-borderGray shadow hover:bg-borderGray hover:shadow-lg duration-150"
  >
    Uuesti
  </button>
</div>
`;
